lib/internal/modules/cjs/loader.js

```diff
@@ -562,6 +562,21 @@ Module._resolveFilename = function(request, parent, isMain, options) {
     return request;
   }
 
+  const parentFilename = parent && parent.filename;
+  const cache = Module._resolveFilename.cache;
+
+  let filename;
+  // do not use the cache
+  // 1. require.resolve('you/request', {paths: []})
+  // 2. don't have parent.filename
+  // 3. don't have cache for this parent.filename
+  if (!(options && options.paths) &&
+      !parentFilename &&
+      cache.get(parentFilename) !== undefined) {
```

Comment:
Doing the `.get()` lookup twice is wasteful. I'd get and store the value first and *then* check if it's `undefined`.

Reply:
Thanks, I'll modify this.